packages/firebase_messaging/README.md

```diff
@@ -76,61 +73,55 @@ By default background messaging is not enabled. To handle messages in the backgr
-            final String key = FirebaseCloudMessagingPluginRegistrant.class.getCanonicalName();
+        private fun alreadyRegisteredWith(registry: PluginRegistry): Boolean {
+            val key = FirebaseCloudMessagingPluginRegistrant::class.java.canonicalName
             if (registry.hasPlugin(key)) {
-                return true;
+                return true
             }
-            registry.registrarFor(key);
-            return false;
+            registry.registrarFor(key)
+            return false
         }
     }
   ```
-1. Add an `Application.java` class to your app in the same directory as your `MainActivity.java`. This is typically found in `<app-name>/android/app/src/main/java/<app-organization-path>/`.
-
-  ```java
-    package io.flutter.plugins.firebasemessagingexample;
-    
-    import io.flutter.app.FlutterApplication;
-    import io.flutter.plugin.common.PluginRegistry;
-    import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback;
-    import io.flutter.plugins.GeneratedPluginRegistrant;
-    import io.flutter.plugins.firebasemessaging.FirebaseMessagingPlugin;
-    import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService;
-
-    public class Application extends FlutterApplication implements PluginRegistrantCallback {
-      @Override
-      public void onCreate() {
-        super.onCreate();
-        FlutterFirebaseMessagingService.setPluginRegistrant(this);
-      }
-
-      @Override
-      public void registerWith(PluginRegistry registry) {
-        FirebaseCloudMessagingPluginRegistrant.registerWith(registry);
-      }
+1. Add an `Application.kt` class to your app in the same directory as your `MainActivity.kt`. This is typically found in `<app-name>/android/app/src/main/kotlin/<app-organization-path>/`.
+
+  ```kt
+    import io.flutter.app.FlutterApplication
+    import io.flutter.plugin.common.PluginRegistry
+    import io.flutter.plugin.common.PluginRegistry.PluginRegistrantCallback
+    import io.flutter.plugins.firebasemessaging.FlutterFirebaseMessagingService
+
+
+    class Application : FlutterApplication(), PluginRegistrantCallback {
+        override fun onCreate() {
+            super.onCreate()
+            FlutterFirebaseMessagingService.setPluginRegistrant(this)
+        }
+
+        override fun registerWith(registry: PluginRegistry) {
+            FirebaseCloudMessagingPluginRegistrant.registerWith(registry)
+        }
     }
   ```
 
-1. In `Application.java` and `FirebaseCloudMessagingPluginRegistrant.java`, make sure to change `package io.flutter.plugins.firebasemessagingexample;` to your package's identifier. Your package's identifier should be something like `com.domain.myapplication`.
+1. In `Application.kt` and `FirebaseCloudMessagingPluginRegistrant.kt`, make sure to change `package io.flutter.plugins.firebasemessagingexample;` to your package's identifier. Your package's identifier should be something like `com.domain.myapplication`.
 
-   ```java
+   ```kt
```

Comment:
There is no `package io.flutter.plugins.firebasemessagingexample;` in `Application.kt` and `FirebaseCloudMessagingPluginRegistrant.kt` code above

Reply:
Dont know how that disappeared. thanks a lot. I know they wont merge this pull request + i dont use flutter anymore. I'm just going to close this request :/